src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprintAction.java

```diff
@@ -0,0 +1,94 @@
+package org.jenkinsci.plugins.docker.commons.fingerprint;
+
+import hudson.model.Fingerprint;
+import hudson.model.Run;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import jenkins.model.RunAction2;
+import org.jenkinsci.plugins.docker.commons.Messages;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+/**
+ * Adds a link, which provides a list of fingerprints with {@link DockerRunPtrFingerprintFacet}s.
+ * @author Oleg Nenashev
+ */
+public class DockerFingerprintAction implements RunAction2 {
+
+    private final Set<String> fingerprintKeys;
+    transient @CheckForNull Run<?, ?> run;
+
+    public DockerFingerprintAction(Run<?, ?> run) {
+        this.run = run;
+        this.fingerprintKeys = new HashSet<String>();
+    }
+
+    public Run<?, ?> getRun() {
+        return run;
+    }
+    
+    @Override
+    public void onAttached(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> r) {
+        run = r;
+    }
+
+    @Override
+    public String getIconFileName() {
+        return "fingerprint.png";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return Messages.DockerFingerprintAction_DisplayName();
+    }
+
+    @Override
+    public String getUrlName() {
+        return "docker-fingerprints";
+    }
+
+    public Set<String> getFingerprintKeys() {
+        return fingerprintKeys;
```

Comment:
Wise to have the `public` method return `Collections.immutableSet` (if not a copy-on-write implementation), and access the field directly from `addToRun`. I suppose that this method is considered an API that other plugins might need.